documentation/src/main/asciidoc/user_guide/query.adoc

```diff
@@ -92,73 +144,468 @@ for (Object book : objectList) {
+declare upfront which types are going to be indexed. This can be done via xml:
 
 [source,xml]
 ----
 <infinispan>
    <cache-container default-cache="default">
-      <local-cache name="default">
-         <indexing index="LOCAL">
-            <property name="default.directory_provider">ram</property>
+      <replicated-cache name="default">
+         <indexing index="ALL">
+            <indexed-entities>
+                <indexed-entity>org.infinispan.query.test.Car</indexed-entity>
+                <indexed-entity>org.infinispan.query.test.Truck</indexed-entity>
+            </indexed-entities>
          </indexing>
-      </local-cache>
+      </replicated-cache>
    </cache-container>
 </infinispan>
 
 ----
 
-In this example the index is stored in memory, so when this nodes is shutdown the index is lost: good for a quick demo, but in real world cases you'll want to use the default (store on filesystem) or store the index in Infinispan as well. For the complete reference of properties to define, refer to the link:http://docs.jboss.org/hibernate/stable/search/reference/en-US/html_single/#search-configuration[Hibernate Search documentation] .
+or programmatically:
+
+[source,java]
+----
+ cacheCfg.indexing()
+      .index(Index.ALL)
+       .addIndexedEntity(Car.class)
+       .addIndexedEntity(Truck.class)
+
+----
+
+
+===== Index mode [[query.index-mode]]
+
+An Infinispan node typically receives data from two sources: local and remote. Local translates to clients manipulating data using the map API in the same JVM;
+remote data comes from other Infinispan nodes during replication or rebalancing.
+
+The index mode configuration defines, from a node in the cluster point of view, which data gets indexed.
+
+Possible values:
+
+* ALL: all data is indexed, local and remote.
+* LOCAL: only local data is indexed.
+* PRIMARY_OWNER: Only entries containing keys that the node is primary owner will be indexed, regardless of local or remote origin.
+* NONE: no data is indexed. Equivalent to not configure indexing at all.
+
+===== Index Managers [[query.index-manager]]
+
+Index managers are central components in Infinispan Querying responsible for the indexing configuration, distribution and internal lifecycle of several query components such as Lucene's
+_IndexReader_ and _IndexWriter_. Each Index Manager is associated with one of more types of _Directory Provider_, which defines the physical storage of the index.
+
+Regarding index distribution, Infinispan can be configured with shared or non-shared indexes.
+
+===== Shared indexes [[query.shared-index]]
+
+A shared index is a single, distributed, cluster-wide index for a certain cache. The main advantage is that the index is visible from every node and can be queried as if the index were local, there is no need to <<query.clustered-query-api,broadcast>>
```

Comment:
Not sure, I think `were` sounds better :smile:
Let's ask a native english speaker, wdyt @ryanemerson, "as if the index were local"  or "as if the index was local" ?